sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -288,25 +291,59 @@ case class AdaptiveSparkPlanExec(
       addSuffix,
       maxFields,
       printNodeId)
-    currentPhysicalPlan.generateTreeString(
+    plans.zipWithIndex.foreach { case ((name, plan), i) =>
```

Comment:
Would it make sense to wrap each plan with a dummy `UnaryExecNode` and call the existing `generateTreeString` method on these dummy operators instead of adding this new `generateTreeStringWithName` method?

Reply:
Since there are always only two plans. Shall we just call `initialPlan.generateTreeString`  and `currentPhysicalPlan.generateTreeString`  here?

Reply:
What @allisonwang-db tries to do here is to print this "CurrentPlan" and "InitialPlan" as a "header" of each plan.

My question was: are we OK with this output style? Alternatively we could also do:
```
AdaptiveSparkPlanExec
:- CurrentPlan: (top node of current plan starts here)
+- InitialPlan: (top node of initial plan starts here)
```

Reply:
+1!

The current format in this PR is a bit misleading. It makes people think that `CurrentPlan` and `InitialPlan` are new nodes.

Reply:
I tried wrapping the plan with a dummy node but it won't work with having the header printed on the same line as the top node without changing the `generateTreeString` method itself. The suggestion makes sense, will update the code.